mlflow/pyfunc/scoring_server/client.py

```diff
@@ -1,9 +1,35 @@
 import requests
 import time
 import json
+import numpy as np
+import pandas as pd
 
 from mlflow.pyfunc import scoring_server
-from mlflow.utils.proto_json_utils import _DateTimeEncoder
+
+from mlflow.exceptions import MlflowException
+
+
+class ScoringServerResponse(dict):
+    """Predictions returned from the MLflow scoring server. The predictions are presented as a
+    dictionary with convenience methods to access the predictions parsed as higher level type."""
+
+    @classmethod
+    def from_raw_json(cls, json_str):
+        try:
+            parsed_response = json.loads(json_str)
+        except Exception as ex:
+            raise MlflowException(f"Model response is not a valid json. Error {ex}")
+        if not isinstance(parsed_response, dict) or "predictions" not in parsed_response:
+            raise MlflowException(
+                "Invalid predictions data. Prediction object must be a dictionary "
+                "with 'predictions' field."
+            )
+        return ScoringServerResponse(parsed_response)
+
+    def get_predictions_dataframe(self):
+        """Get the predictions returned from the server as pandas.DataFrame. this method will fail
+        if the returned predictions is not a valid"""
+        return pd.DataFrame(data=self["predictions"])
```

Comment:
Following from https://github.com/mlflow/mlflow/pull/6575/files#r959181856, is this class necessary? It doesn't support the full set of pyfunc outputs

Reply:
Responded above.